pkg/controller/v1beta1/inferenceservice/controller_test.go

```diff
@@ -220,6 +221,7 @@ var _ = Describe("v1beta1 inference service controller", func() {
 									"key3":                                "val3FromPredictor",
 								},
 								Annotations: map[string]string{
+									"serving.kserve.io/deploymentMode":                         "Serverless",
```

Comment:
I guess we need to add this here because in the test we did not set the `DefaultDeploymentMode` from config right?

Reply:
In this test the inference services weren't created with a `deployConfig` as they weren't created using [`SetupWithManager()`](https://github.com/kserve/kserve/blob/master/pkg/controller/v1beta1/inferenceservice/controller.go#L277), so I added this annotation as a workaround. Without the deployment mode, `RoutesReady` status wouldn't be set.